lib/aytests/cli.rb

```diff
@@ -42,11 +42,19 @@ def test(*files)
         help("test")
         exit TEST_DOES_NOT_EXIST_ERRCODE
       end
-
       results = files.each_with_object({}) do |test_file, hsh|
         hsh[test_file] = test_result(test_file)
       end
       show_results(results)
+      tests_again = results.select {|file,ret| ret == :failed }.keys
+      unless tests_again.empty?
+        say "giving following tests an additional try:"
+        say tests_again
+        tests_again.each do |test_file|
+          results[test_file] = test_result(test_file)
+        end
+      end
+      show_results(results) unless tests_again.empty?
```

Comment:
What about calling `show_results` only once (without condition)? Just remove the first call and the condition of this one.